rocAL/rocAL/source/mxnet_meta_data_reader.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright (c) 2019 - 2020 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <iostream>
+#include <utility>
+#include <algorithm>
+#include <memory.h>
+#include <stdint.h>
+#include "mxnet_meta_data_reader.h"
+
+using namespace std;
+
+void MXNetMetaDataReader::init(const MetaDataConfig &cfg)
+{
+    _path = cfg.path();
+    _output = new LabelBatch();
+}
+
+bool MXNetMetaDataReader::exists(const std::string& _image_name)
+{
+    return _map_content.find(_image_name) != _map_content.end();
+}
+
+void MXNetMetaDataReader::add(std::string image_name, int label)
+{
+    pMetaData info = std::make_shared<Label>(label);
+    if(exists(image_name))
+    {
+        WRN("Entity with the same name exists")
+        return;
+    }
+    _map_content.insert(std::pair<std::string, std::shared_ptr<Label>>(image_name, info));
+}
+
+void MXNetMetaDataReader::lookup(const std::vector<std::string> &_image_names)
+{
+    if(_image_names.empty())
+    {
+        WRN("No image names passed")
+        return;
+    }
+    if(_image_names.size() != (unsigned)_output->size())   
+        _output->resize(_image_names.size());
+
+    for(unsigned i = 0; i < _image_names.size(); i++)
+    {
+        auto _image_name = _image_names[i];
+        auto it = _map_content.find(_image_name);
+        if(_map_content.end() == it)
+            THROW("ERROR: Given name not present in the map"+ _image_name )
+        _output->get_label_batch()[i] = it->second->get_label();
+    }
+}
+
+void MXNetMetaDataReader::print_map_contents()
+{
+    std::cerr << "\nMap contents: \n";
+    for (auto& elem : _map_content) {
+        std::cerr << "Name :\t " << elem.first << "\t ID:  " << elem.second->get_label() << std::endl;
+    }
+}
+
+void MXNetMetaDataReader::read_all(const std::string &path)
+{
+    std::string _rec_file, _idx_file;
+    if (_path.find("train") != std::string::npos)
+    {
+        _rec_file = _path + "/train.rec";
```

Comment:
Is the record name going to be same for all datasets. It might be better to get read the .rec and .idx files present in the directory instead of assuming fixed names

Reply:
Removed fixed names & read .rec and .idx files in the given directory.